libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteOptions.kt

```diff
@@ -0,0 +1,21 @@
+package com.mapbox.navigation.core.fasterroute
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+
+@ExperimentalPreviewMapboxNavigationAPI
+data class FasterRouteOptions internal constructor(
+    val maxAcceptableGeometrySimilarityToRejectedAlternatives: Double
+) {
+    class Builder {
+
+        private var maxGeometrySimilarityToRejectedAlternatives = 0.5
```

Comment:
This ratio might work for intercity routes (or longer) but might make us unable to suggest faster alternatives in city centers that often rely on taking a turn earlier to later rejoin the original route and avoid congested intersection. These types of routes could have both a significant overlap while still having a significant time delta. I'm just speculating but just wanted to mention this as a discussion point, is there a more complex metric that we could use to account for these situations? Maybe road classes could be taken into account.

Reply:
> Maybe road classes could be taken into account.

Do you know to find road class of a step?

Reply:
> make us unable to suggest faster alternatives in city centers that often rely on taking a turn earlier to later rejoin the original route and avoid congested intersection

We do not compare alternative with the primary route. But if there is a rejected alternative which is very similar to the primary route, it could be a problem. 

What if we consider initial alternatives as rejected only if user picks slower route? It will decrease chance of this happening.

Reply:
> is there a more complex metric that we could use to account for these situations?

I will try adding comparison of steps names